apps/dotcom/sync-worker/src/TLUserDurableObject.ts

```diff
@@ -360,7 +371,11 @@ export class TLUserDurableObject extends DurableObject<Environment> {
 							await tx.updateTable(update.table).set(updates).where('id', '=', id).execute()
 							if (update.table === 'file') {
 								const currentFile = this.cache.store.getFullData()?.files.find((f) => f.id === id)
-								if (currentFile && currentFile.published !== rest.published) {
```

Comment:
This is another subtle error. We only relied on the published field to be different. But `rest` is a partial, so it might not even have the `published` field set. So we also need to check that we did receive some value here. Similar for the `lastPublished`, otherwise we'd compare a number to undefined.